src/app/common-components/bread-crumbs/bread-crumbs.component.ts

```diff
@@ -1,14 +1,18 @@
-import { Component, Input } from "@angular/core";
+import { Component, Input, ChangeDetectionStrategy, OnInit } from "@angular/core";
+import { AppService } from "src/app/app.service";
 
 @Component({
     selector: "app-bread-crumbs",
     templateUrl: "./bread-crumbs.component.html",
     styleUrls: ["./bread-crumbs.component.less"]
 })
-export class BreadCrumbsComponent {
+export class BreadCrumbsComponent implements OnInit {
     public completePath = "";
 
-    @Input() public set path(path) {
-        this.completePath = path.join("/ ");
+    constructor(private appService: AppService) {}
+    ngOnInit() {
+        this.appService.breadCrumbs$.subscribe(path => {
```

Comment:
Do not forget to unsubscribe in ngOnDestroy()

Or, and more preferable to use async pipe in the template.
So, you can create public breadCrumbs$: Observable<string>
and assign it in ngOnInit():
this.breadCrumbs$ = this.appService.breadCrumbs$
also, path && path.join(" / ") logic should be in the service.
All the logic should be placed in services as much as it possible.
That's why we use services - to separate view and logic